docs/udp.rst

```diff
@@ -34,79 +34,188 @@ This turns your openEO process into a web application
 that can be executed using the regular openEO
 support for synchronous and asynchronous jobs.
 
-
 Process Parameters
 ====================
 
-User-defined processes are usually **parameterized**,
-meaning certain inputs are expected when calling the process.
+User-defined processes are usually **parameterized**, 
+meaning specific inputs might be needed when calling the process. 
+It allows the user to perform specific tasks based on the provided inputs.
+
+For example, let us consider that we want to recalculate the EVI-timeseries
+as done in :ref:`example-use-case-evi-map-and-timeseries` but for a different area.
+Then we can either redefine the whole process or simply define it as a UDP ``EVI_timeseries``, 
+that can be easily shared and reused without dealing with the entire workflow each time.
+
+
+So, let's define temporal_extent,spatial_extent, and 
+geometry/polygon used for pixel aggregation as parameters. 
+The resulting code block will appear as follows:
+
+.. code-block:: python
+
+    # Load data collection
+    sentinel2_cube = connection.load_collection(
+        "SENTINEL2_L2A",
+        spatial_extent = polygon,
+        temporal_extent = temporal_interval,
+        bands = ["B02", "B04", "B08", "SCL"],
+    )
+    # Extract spectral bands and calculate EVI with the "band math" feature
+    blue = sentinel2_cube.band("B02") * 0.0001
+    red = sentinel2_cube.band("B04") * 0.0001
+    nir = sentinel2_cube.band("B08") * 0.0001
+    evi = 2.5 * (nir - red) / (nir + 6.0 * red - 7.5 * blue + 1.0)
+
+    # Use the scene classification layer to mask out non-vegetation pixels
+    scl = sentinel2_cube.band("SCL")
+    evi_masked = evi.mask(scl != 4)
+
+    evi_aggregation = evi_masked.aggregate_spatial(
+        geometries = feature,
+        reducer = "mean",
+    )
+
 
-For example, if you often have to convert Fahrenheit to Celsius::
+Additionally, based on our application, we can also define the collection,
+or even bands as parameters as shown below:
 
-    c = (f - 32) / 1.8
+.. code-block:: python
 
-you could define a user-defined process ``fahrenheit_to_celsius``,
-consisting of two simple mathematical operations
-(pre-defined processes ``subtract`` and ``divide``).
+    datacube = connection.load_collection(
+            collection_id = collection,
+            temporal_extent = temporal_interval
+            spatial_extent = polygon,
```

Comment:
again it is not clear what these variables `collection`, `temporal_interval` ... are